vscode/src/statusbar.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { log } from "qsharp-lang";
+import * as vscode from "vscode";
+import { isQsharpDocument } from "./common";
+
+export function activateTargetProfileStatusBarItem(): vscode.Disposable[] {
+  const disposables = [];
+
+  disposables.push(registerTargetProfileCommand());
+
+  const statusBarItem = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Right,
+    200
+  );
+  disposables.push(statusBarItem);
+
+  statusBarItem.command = "quantum-set-target";
+
+  disposables.push(
+    vscode.window.onDidChangeActiveTextEditor((editor) => {
+      if (editor && isQsharpDocument(editor.document)) {
+        refreshStatusBarItemValue();
+      } else {
+        // Could still be showing if the document language was changed away from Q#
+        statusBarItem.hide();
+      }
+    })
+  );
+
+  disposables.push(
+    vscode.workspace.onDidChangeConfiguration((event) => {
+      if (
+        vscode.window.activeTextEditor &&
+        isQsharpDocument(vscode.window.activeTextEditor.document) &&
+        event.affectsConfiguration("qsharp.targetProfile")
+      ) {
+        refreshStatusBarItemValue();
+      }
+    })
+  );
+
+  if (
+    vscode.window.activeTextEditor &&
+    isQsharpDocument(vscode.window.activeTextEditor.document)
+  ) {
+    refreshStatusBarItemValue();
+  }
+
+  function refreshStatusBarItemValue() {
+    // The target profile setting is a "window" scoped setting,
+    // meaning it can't be set on a per-folder basis. So we don't
+    // need to pass a specific scope here to retrieve the document
+    // value - we just use the workspace setting.
+    // VS Code will return the default value defined by the extension
+    // if none was set by the user, so targetProfile should always
```

Comment:
This is true if you have the configuration in the package.json and thus it contributes to the VS Code settings UX. I wonder if we need that UX though, being we have a command and status bar item to change it with anyway. Once Azure is on by default, we'd have no other settings in the UX (yet).

So maybe remove it from package.json for a cleaner UX? (And thus default value would be undefined).

Reply:
I'm not sure where we would persist it then. Can we just persist arbitrary configurations not defined in package.json? I wasn't sure of that.

Reply:
Actually, I've been testing, and if you remove the 'configuration' contribution from package.json, while you can still read any arbitrary setting, you can't actually write one that's not defined!! So we need to leave this in package.json.

I renamed to "Q#.targetProfile" and integrated with my branch. If you check out my azure PR, this with changes is in there and works great!